lib/groupio.c

```diff
@@ -313,7 +313,6 @@ static /*@null@*/struct commonio_entry *merge_group_entries (
 	new_line_len = strlen (gr1->line) + strlen (gr2->line) +1;
 	new_line = (char *)malloc (new_line_len + 1);
 	if (NULL == new_line) {
-		errno = ENOMEM;
```

Comment:
Hm, but the manpage doesn't guarantee that for malloc()

Reply:
(I'm probably ok with it, just curious what you have to say about it :-)

Reply:
I guess you refer to this?:

`malloc(3)`:
```
   Nonportable behavior
       The behavior of these functions when the  requested  size  is  zero  is
       glibc  specific;  other implementations may return NULL without setting
       errno, and portable POSIX programs should tolerate such behavior.   See
       realloc(3p).

       POSIX  requires  memory allocators to set errno upon failure.  However,
       the C standard does not require this, and applications portable to non-
       POSIX platforms should not assume this.
```

If so, POSIX requires setting errno on error.

Another thing is what happens when we try to allocate 0 bytes, which I hope we don't do, but even if we do, I'm not convinced it's an error, and POSIX isn't either:

`malloc(3posix)`:
```
RETURN VALUE
       Upon successful completion with size not equal to 0, malloc() shall re‐
       turn a pointer to the allocated space. If size is 0, either:

        *  A  null pointer shall be returned and errno may be set to an imple‐
           mentation-defined value, or

        *  A pointer to the allocated space shall be returned. The application
           shall ensure that the pointer is not used to access an object.

       Otherwise, it shall return a null pointer and set errno to indicate the
       error.
```

Reply:
I'm just referring to the fact that the manpage does explicitly say that reallocarray sets errno to ENOMEM but does not say that about malloc.

But clearly a local testcase proves that it does, and so as long as you're comfortable with it please feel free to resolve this :)

Reply:
Maybe I have a too new manual page; can you please quote what you're seeing?  Mine says:

```
RETURN VALUE
       The malloc(), calloc(), realloc(), and reallocarray() functions  return
       a  pointer  to  the allocated memory, which is suitably aligned for any
       type that fits into the requested size or less.  On error, these  func‐
       tions  return  NULL  and  set  errno.
```

Reply:
> But clearly a local testcase proves that it does, and so as long as you're comfortable with it please feel free to resolve this :)

Yeah, I am.  I am rather confused about the manual page itself, and maybe I need to fix some inconsistency there :)

Reply:
Hm, I replied in email, but I don't see the reply here.  In any case - I've found text that does say errno will be set, so I'll resolve this, thanks.

Reply:
Email replies don't go threaded (github sucks in many ways).  I answered below :)